bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -508,29 +509,130 @@ public Void call() throws Exception {
-            logger.error(
-                    "Exception occured while calling thing handler factory '" + thingHandlerFactory + "': " + message,
-                    ex.getCause());
         } catch (TimeoutException ex) {
             ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
-                    ThingStatusDetail.HANDLER_INITIALIZING_ERROR, ex.getMessage());
+                    ThingStatusDetail.HANDLER_REGISTERING_ERROR,
+                    ex.getCause() != null ? ex.getCause().getMessage() : ex.getMessage());
             setThingStatus(thing, statusInfo);
             logger.warn("Registering handler for thing '{}' takes more than {}ms.", thing.getUID(),
                     SafeMethodCaller.DEFAULT_TIMEOUT);
         } catch (Exception ex) {
             ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
-                    ThingStatusDetail.HANDLER_INITIALIZING_ERROR, ex.getMessage());
+                    ThingStatusDetail.HANDLER_REGISTERING_ERROR,
+                    ex.getCause() != null ? ex.getCause().getMessage() : ex.getMessage());
             setThingStatus(thing, statusInfo);
             logger.error("Exception occured while calling thing handler factory '" + thingHandlerFactory + "': "
                     + ex.getMessage(), ex);
         }
     }
 
+    private void initializeHandler(Thing thing) {
+        if (isInitializable(thing)) {
+            ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.INITIALIZING, ThingStatusDetail.NONE);
+            setThingStatus(thing, statusInfo);
+            initializeHandler(thing, thing.getHandler());
+        } else {
+            logger.debug("Thing '{}' not initializeable, check required configuration parameters.", thing.getUID());
+            ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
+                    ThingStatusDetail.HANDLER_CONFIGURATION_PENDING);
+            setThingStatus(thing, statusInfo);
+        }
+    }
+
+    private boolean isInitializable(Thing thing) {
+        // determines if all 'required' configuration parameters are available in the configuration
+        logger.debug("Determine if thing '{}' is initializable.", thing.getUID());
+
+        ThingType thingType = TypeResolver.resolve(thing.getThingTypeUID());
+        if (thingType != null) {
+            ConfigDescription description = TypeResolver.resolve(thingType.getConfigDescriptionURI());
+            if (description == null) {
+                return true;
+            }
+
+            List<String> requiredParameters = getRequiredParameters(description);
+            Map<String, Object> properties = thing.getConfiguration().getProperties();
+            return properties.keySet().containsAll(requiredParameters);
+        } else {
+            logger.debug("Thing type '{}' of thing '{}' not found.", thing.getThingTypeUID(), thing.getUID());
+        }
+        return false;
```

Comment:
You are right. The 'GenericThingProvider' creates a Thing, also without a type in the 'ThingTypeRegistry'.

Hence: a Thing is always initializable if a ThingType or a ConfigDescription can not be resolved. If a ConfigDescription is available, the Thing configuration must match the required parameter.